irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -156,40 +162,18 @@ namespace {
 namespace iroha {
   namespace ametsuchi {
 
-    std::string CommandError::toString() const {
-      return (boost::format("%s: %s") % command_name % error_message).str();
-    }
-
-    PostgresCommandExecutor::PostgresCommandExecutor(soci::session &sql)
-        : sql_(sql), do_validation_(true) {}
-
-    void PostgresCommandExecutor::setCreatorAccountId(
-        const shared_model::interface::types::AccountIdType
-            &creator_account_id) {
-      creator_account_id_ = creator_account_id;
-    }
-
-    void PostgresCommandExecutor::doValidation(bool do_validation) {
-      do_validation_ = do_validation;
-    }
-
-    CommandResult PostgresCommandExecutor::operator()(
-        const shared_model::interface::AddAssetQuantity &command) {
-      auto &account_id = creator_account_id_;
-      auto &asset_id = command.assetId();
-      auto amount = command.amount().toStringRepr();
-      auto precision = command.amount().precision();
-      boost::format cmd(R"(
+    const std::string PostgresCommandExecutor::addAssetQuantityBase = R"(
+          PREPARE %s (text, text, int, text) AS
           WITH has_account AS (SELECT account_id FROM account
-                               WHERE account_id = :account_id LIMIT 1),
+                               WHERE account_id = $1 LIMIT 1),
```

Comment:
Why named aliases replaced with anonymous ids? It's harder to read/undertstand the query and more error-prone

Reply:
Because postgres understands only positional arguments. Previously we substituted parameters in our code, now it is done as a prepared statement

Reply:
Got it, I though it was soci args, thanks!